feat(clayui.com): Document using Clay in JSPs

kresimir-coko opened the pull request:
Hey @bryceosterhaus this PR addresses #3286, and it aims to fill some initial documentation requirements for our users that want to use Clay in JSPs.

### Some notes
-  I took inspiration from [Liferay Help Center](https://help.liferay.com/hc/en-us/articles/360017888252-Using-the-Clay-Taglib-in-Your-portlets) when it comes to text
-  I referenced [liferay-clay.tld](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld) to get the list of components that exist as tags
    -  Badge, Icon, Label & Sticker are for some reason marked as deprecated, making me think that Clay doesn't support them 😕 
    -  Stripe & Sheet are mentioned but we don't have them on clayui.com 🤷

matuzalemsteles commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
```diff
@@ -0,0 +1,70 @@
+</div>
+</div>
+
+Clay in combination with DXP provides a set of tags for creating Clay components in your JSP files.
```
I would like to make things more explicit here that the version contained in the JSP are v2 APIs and that it uses v2 components and it would be great to make it clear that the JSPs are only available in Liferay DXP, a link to the product would be great too.

matuzalemsteles replied:
Just to know, I didn't follow the taglib changes, are those listed using the components in v3?

jbalsas replied:
The ones listed follow the v3 APIs.

Just to clarify, many of these tags DO NOT USE the React Components. They simply render basic markup and mimic the React API since we don't have a stable and performant way of rendering them from Java.

matuzalemsteles replied:
Thanks @jbalsas for clarifying.

jbalsas commented:
Hey @kresimir-coko, where did you see the tags are deprecated? 

Only specific attributes of the tag are marked as deprecated. Those are the ones that belong to the `2.x` version of the tag, that we have deprecated moving forward. 

To avoid confusion, we should only document those related to `3.x`.

Just to clarify, those tags are NOT deprecated 😉

matuzalemsteles commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
```diff
@@ -0,0 +1,70 @@
+
+Clay in combination with DXP provides a set of tags for creating Clay components in your JSP files.
+
+## Setting up your module
```
I have a little bit of concern about it, @mwilliams2014 has done a lot of work documenting the use of JSP for Clay v2 components and I am concerned that this may become more difficult to maintain because the code is on Liferay and that we may have duplicate information, in theory, the next updates should "follow" the component APIs so there would be no problem but I still wonder if we should just link to the Liferay developer documentation.

Another concern is that the JSP comes with Liferay DXP and depending on the version that the person is using the DXP it may be that the JSP or API he is seeing here is not available or the APIs are only available in the latest version of DXP or on the master itself.

So what I want to say is that we should be concerned with this before going ahead and aligning it if we want these documents to be only for internal DXP developers and keep only updated according to the master or we would have to follow the DXP version and update only after releases. If it is for internal developers, we must make this clear and also the person if they are looking for a specific version or if it is a customer or partner forwarding to the Liferay DXP documentation.

cc @jbalsas @wincent

mwilliams2014 replied:
Thanks for the mention @matuzalemsteles . We definitely don't want to duplicate information in both places. If you have documentation that's for our customers, that relates directly to its use in Liferay, please draft it and send it to me for review, so we can publish it with/improve our current documentation. If it's meant for internal usage/is clay specific (not relating to Liferay) then it makes more sense to publish that information with Clay's documentation.

As far as Liferay features that are DXP only, we tend to note that above the documentation along with the version (i.e. DXP 7.3 SP2+), so users are aware of that, and update just before/after releases as you mentioned. Thanks!

bryceosterhaus commented:
Yeah this seems like a slippery slope for us in how we document or communicate jsp and portal specific docs. I'm fine to include jsp code snippets in our docs, but I'm not sure we want to be maintaining docs on setting up modules and how to use taglibs in portal. It also probably isnt helpful for devs to look for docs on both our portal docs site and then also having to come to clay.

@jbalsas how far into the portal world do you think we should document things? Is just having snippets without much explanation okay?

kresimir-coko commented:
@jbalsas:
> where did you see the tags are deprecated?
Oh, I didn't understand that the deprecate description was related to an attribute, don't have much xml experience. I'll make sure to keep that in mind when/if we pick this issue up again.

jbalsas commented:
> [...] It also probably isnt helpful for devs to look for docs on both our portal docs site and then also having to come to clay.

I have to yet meet an internal dev that dives in our portal docs...

> [...] how far into the portal world do you think we should document things? Is just having snippets without much explanation okay?

Yeah, that's what I suggested, just the tld import and how to use the jsps as if they were components

> We definitely don't want to duplicate information in both places. If you have documentation that's for our customers, that relates directly to its use in Liferay, please draft it and send it to me for review, so we can publish it with/improve our current documentation. If it's meant for internal usage/is clay specific (not relating to Liferay) then it makes more sense to publish that information with Clay's documentation.

Take not of what @mwilliams2014 said here. We're targeting our product team developers here, not our customers. We need to focus on what they need to get their job done. This is usually common snippets and what different attributes they can use to get something done.

kresimir-coko commented:
Hey @bryceosterhaus I've added the links to components that I initially thought were deprecated and changed some wording. Not sure if there's anything more that our devs would need to get started.

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
```diff
@@ -0,0 +1,74 @@
+
+To use Clay components in your JSP apps, add the following to your module:
+
+### Gradle
```
I think we might be better off leaving out this gradle portion since this more falls under dxp specific API and the more we can focus on just the tags, the better.

kresimir-coko replied:
@jbalsas what do you think about this? 

My thinking is that anyone who comes looking for Clay in JSPs is gonna appreciate this information.

jbalsas replied:
I agree with @bryceosterhaus.

This is a build setup that's unique to DXP. You have customers and even Workspace posibly using Maven setups. We might move to another build system...  chances are the gradle setup already includes the dependency...

Let's stick to the JSP here.

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
````diff
@@ -0,0 +1,74 @@
+<%@ taglib prefix="clay" uri="http://liferay.com/tld/clay" %>
+```
+
+## Using Clay taglib in your JSPs
````
I don't think we need this header, it can all fall under `## Using Taglibs`

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
````diff
@@ -0,0 +1,74 @@
+<clay:button label="<%= Button Label %>" style="primary" />
+```
+
+## Clay components available in JSPs
````
```suggestion
## Clay taglibs available
```

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
```diff
@@ -0,0 +1,74 @@
+
+Clay taglibs provide the following UI components for your apps:
+
+-   [Alert](/docs/components/alert.html)
+-   [Badge](/docs/components/badge.html)
+-   [Button](/docs/components/button.html)
+-   [Card](/docs/components/card.html)
+-   [Checkbox](/docs/components/checkbox.html)
+-   [Dropdown](/docs/components/drop-down.html)
+-   [Form Elements](/docs/components/form.html)
+-   [Icon](/docs/components/icon.html)
+-   [Label](/docs/components/label.html)
+-   [Layout Elements](/docs/components/layout.html)
+-   [Link](/docs/components/link.html)
+-   [Management Toolbar](/docs/components/management-toolbar.html)
+-   [Multi Select](/docs/components/multi-select.html)
+-   [Navigation Bar](/docs/components/navigation-bar.html)
+-   [Progress Bar](/docs/components/progress-bar.html)
+-   [Radio](/docs/components/radio.html)
+-   [Select](/docs/components/select.html)
+-   [Sticker](/docs/components/sticker.html)
+-   [Table](/docs/components/table.html)
```
We should probably only reference the component here once we get the taglib snippets in

Another option is to make this page a draft for now by setting `draft: true` in the frontmatter and then make it live once we add all the snippets.

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
````diff
@@ -0,0 +1,74 @@
+compileOnly project(":apps:frontend-taglib:frontend-taglib-clay")
+```
+
+### JSP
````
```suggestion
### Using Taglibs
```

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
```diff
@@ -0,0 +1,74 @@
+
+Clay in combination with DXP provides a set of tags for creating Clay components in your JSP files.
+
+## Setting up your module
```
I think we should just scrap the whole "module" heading

bryceosterhaus commented on clayui.com/content/docs/get-started/using-clay-in-jsps.md:
```diff
@@ -0,0 +1,74 @@
+<div class="nav-toc-absolute">
+<div class="nav-toc">
+
+-   [Setting up your module](#setting-up-your-module)
+-   [Using Clay taglib in your JSPs](#using-clay-taglib-in-your-jsps)
+-   [Clay components available in JSPs](#clay-components-available-in-jsps)
```
If we scrap the setting up module and gradle portion, which I think we should so that we don't have to concern ourselves too much with the dxp build specifics, I think we could structure the page like...

```suggestion
-   [Using Taglibs](#using-taglibs)
-   [Clay taglibs available](#clay-taglibs-available)
```

kresimir-coko commented:
@bryceosterhaus I've pushed the latest changes, I removed the Gradle stuff, changed the nav, and made it a draft

bryceosterhaus commented:
Looks great! Thanks!
